### ISSUE TITLE
Asking for preproduce ComplEx on ORIGINAL FB15K

### ISSUE THREAD

**Issue by phucty** [facebookresearch/kbc#10]
Thank you for your time to fine-tune the new parameters.
 
I have a problem with FB15K dataset, could you please correct me if I wrong somewhere? 

**Original FB15K**
I can not reproduce the same result on the **original** FB15K dataset and the dataset that you provide. 
This link: [https://everest.hds.utc.fr/lib/exe/fetch.php?media=en:fb15k.tgz]
The MRR score of Valid set reach 0.84728584 at Epoch 115, and then it started to fall.

**FB15K**
I also tested on the dataset that you provide in this link [https://dl.fbaipublicfiles.com/kbc/data.tar.gz]
(All entities and relations was encoded to integer IDs). 
With the new parameters, the model performance can only reach to	MRR:0.84741566 at Epoch 116 on the valid set.

What should I do to reproduce the MRR score of 0.86 on the FB15K dataset? 

Here this is the code that I used to test. 
**FB15K - ComplEx - Rank 2000 - reg 2.5e-3**
`python kbc/learn.py --dataset FB15K --model ComplEx --rank 2000 --optimizer Adagrad --learning_rate 1e-2 --batch_size 100 --regularizer N3 --reg 2.5e-3 --max_epochs 200 --valid 1 `

This is the log file on the **Original FB15K** dataset:
Epoch	Train(loss)	Valid(MRR)	Test(MRR)
0	0.28557111	0.21563877	0.21695131
1	0.51367164	0.34545964	0.34569839
2	0.71805871	0.48132567	0.47928540
3	0.82222313	0.56841329	0.56578165
4	0.88000029	0.62548071	0.62345359
5	0.91472250	0.66596788	0.66418925
6	0.93698156	0.69591397	0.69439766
7	0.95358196	0.71943289	0.71805435
8	0.96355888	0.73751086	0.73588541
9	0.97203004	0.75185460	0.75069633
10	0.97879598	0.76432011	0.76255512
11	0.98380214	0.77389356	0.77257067
12	0.98672199	0.78240332	0.78084379
13	0.98983303	0.78926352	0.78804612
14	0.99128643	0.79544601	0.79427382
15	0.99359611	0.80076903	0.79957074
16	0.99423641	0.80527973	0.80403480
17	0.99501967	0.80940974	0.80801624
18	0.99618039	0.81269062	0.81145489
19	0.99689016	0.81547362	0.81428304
20	0.99707615	0.81818026	0.81722468
21	0.99751508	0.82045570	0.81961802
22	0.99799338	0.82271165	0.82185829
23	0.99809900	0.82455397	0.82380486
24	0.99842545	0.82617694	0.82553080
25	0.99881667	0.82775342	0.82701689
26	0.99896574	0.82907116	0.82846227
27	0.99898544	0.83034629	0.82978585
28	0.99923363	0.83151683	0.83103910
29	0.99919987	0.83264852	0.83193302
30	0.99923709	0.83367193	0.83296466
31	0.99939093	0.83448964	0.83397171
32	0.99949011	0.83532438	0.83483064
33	0.99951237	0.83608401	0.83568957
34	0.99949709	0.83680180	0.83645040
35	0.99965772	0.83754161	0.83714667
36	0.99962586	0.83809781	0.83781001
37	0.99970272	0.83862516	0.83830112
38	0.99973676	0.83922428	0.83888388
39	0.99973369	0.83966893	0.83941740
40	0.99973398	0.84024203	0.83985305
41	0.99984252	0.84062093	0.84031960
42	0.99974993	0.84110683	0.84075865
43	0.99980173	0.84145269	0.84105837
44	0.99983624	0.84181148	0.84132946
45	0.99983835	0.84220415	0.84167388
46	0.99983218	0.84245965	0.84199527
47	0.99986276	0.84268475	0.84226802
48	0.99981329	0.84295464	0.84251139
49	0.99989858	0.84327799	0.84273136
50	0.99989870	0.84347713	0.84295285
51	0.99994585	0.84364322	0.84318906
52	0.99992108	0.84382129	0.84346214
53	0.99990213	0.84408641	0.84364283
54	0.99988773	0.84425330	0.84381098
55	0.99993330	0.84451291	0.84402552
56	0.99993637	0.84467125	0.84414390
57	0.99996072	0.84483013	0.84425008
58	0.99990815	0.84486419	0.84439784
59	0.99994701	0.84501472	0.84453636
60	0.99995145	0.84514958	0.84472007
61	0.99997666	0.84520793	0.84484276
62	0.99994531	0.84535003	0.84493870
63	0.99991095	0.84543225	0.84513894
64	0.99995717	0.84551516	0.84519318
65	0.99992430	0.84561467	0.84530640
66	0.99996042	0.84570009	0.84536874
67	0.99996126	0.84582907	0.84548032
68	0.99996677	0.84588811	0.84564543
69	0.99995053	0.84600341	0.84571698
70	0.99996054	0.84607205	0.84568727
71	0.99996501	0.84611568	0.84581268
72	0.99997050	0.84620091	0.84589335
73	0.99995700	0.84630886	0.84599790
74	0.99997500	0.84642345	0.84612584
75	0.99991444	0.84646434	0.84616750
76	0.99997142	0.84654942	0.84629130
77	0.99998251	0.84653986	0.84627274
78	0.99994311	0.84658444	0.84634748
79	0.99996251	0.84666705	0.84637389
80	0.99998498	0.84672299	0.84642160
81	0.99995297	0.84675327	0.84646890
82	1.00000000	0.84674850	0.84651664
83	0.99997333	0.84680906	0.84656042
84	0.99994835	0.84689119	0.84655464
85	0.99996999	0.84693623	0.84662983
86	0.99998507	0.84692633	0.84664306
87	0.99998498	0.84696662	0.84668371
88	0.99996778	0.84699124	0.84675461
89	0.99996498	0.84703335	0.84671801
90	0.99994567	0.84704572	0.84679103
91	0.99997625	0.84705746	0.84683430
92	0.99999079	0.84705797	0.84683645
93	0.99996543	0.84706438	0.84691560
94	0.99998000	0.84705409	0.84686515
95	0.99995849	0.84709358	0.84686705
96	0.99997500	0.84713063	0.84687436
97	0.99998042	0.84716102	0.84692174
98	0.99993712	0.84716469	0.84691164
99	0.99997512	0.84724975	0.84698278
100	0.99998999	0.84722605	0.84696332
101	0.99998501	0.84714711	0.84697026
102	0.99999091	0.84714139	0.84696552
103	1.00000000	0.84712300	0.84702221
104	0.99998251	0.84711015	0.84705314
105	1.00000000	0.84714445	0.84705025
106	0.99995998	0.84717971	0.84705117
107	0.99998999	0.84714749	0.84712243
108	0.99996001	0.84717065	0.84712589
109	0.99996001	0.84720722	0.84714031
110	0.99999499	0.84714127	0.84714335
111	0.99998999	0.84718046	0.84716892
112	0.99996015	0.84724343	0.84717712
113	0.99997124	0.84727567	0.84722853
114	0.99998835	0.84727034	0.84719172
115	1.00000000	0.84728584	0.84723872
116	0.99994627	0.84724098	0.84721979
117	0.99997625	0.84724268	0.84721857
118	0.99996498	0.84724936	0.84721971
119	0.99998999	0.84718457	0.84722593
120	0.99997997	0.84718266	0.84721348
121	0.99997723	0.84711304	0.84713510
122	1.00000000	0.84709558	0.84712282
123	0.99998197	0.84714195	0.84721988
124	0.99996257	0.84714207	0.84715748
125	0.99996999	0.84711239	0.84706751
126	0.99994999	0.84713465	0.84712446
127	0.99999031	0.84710392	0.84714404
128	0.99998999	0.84715080	0.84706330
129	1.00000000	0.84712383	0.84707800
130	1.00000000	0.84713739	0.84704885
131	0.99998498	0.84711719	0.84699729
132	0.99997750	0.84713098	0.84701556
133	0.99999499	0.84707537	0.84698561
134	1.00000000	0.84704778	0.84698290
135	0.99998751	0.84702012	0.84699649
136	0.99999031	0.84702533	0.84692785
137	0.99998248	0.84701875	0.84700808
138	0.99999499	0.84704241	0.84696236
139	0.99999055	0.84700993	0.84698820
140	0.99998063	0.84700748	0.84691477
141	0.99997997	0.84703130	0.84689963
142	1.00000000	0.84700713	0.84691343
143	0.99995500	0.84700537	0.84693551
144	0.99998498	0.84695253	0.84687895
145	0.99998501	0.84691501	0.84682265
146	0.99996090	0.84697160	0.84673926
147	0.99998501	0.84695709	0.84674859
148	0.99995625	0.84690955	0.84676823
149	0.99996501	0.84686524	0.84675056
150	0.99998999	0.84684649	0.84672481
151	0.99997538	0.84680137	0.84672895
152	0.99998003	0.84676564	0.84669957
153	0.99995500	0.84676844	0.84673354
154	1.00000000	0.84674770	0.84668726
155	0.99999249	0.84671041	0.84664011
156	0.99998042	0.84669632	0.84662616
157	0.99996001	0.84671518	0.84662899
158	0.99998501	0.84667584	0.84661323
159	1.00000000	0.84656939	0.84654659
160	0.99997646	0.84660426	0.84652206
161	0.99996710	0.84654680	0.84648311
162	0.99998501	0.84651509	0.84647587
163	0.99994299	0.84647438	0.84639782
164	0.99998748	0.84648508	0.84638712
165	0.99996501	0.84642461	0.84640118
166	0.99997997	0.84636709	0.84640270
167	0.99998501	0.84637338	0.84636074
168	0.99997750	0.84631938	0.84632099
169	0.99997500	0.84631246	0.84629214
170	0.99996504	0.84632692	0.84625468
171	0.99998498	0.84628534	0.84626123
172	0.99998498	0.84625691	0.84622723
173	0.99997255	0.84620216	0.84613714
174	0.99997997	0.84617501	0.84614974
175	0.99998999	0.84615323	0.84610152
176	0.99998999	0.84614855	0.84604591
177	0.99999499	0.84608182	0.84602916
178	1.00000000	0.84608290	0.84602734
179	0.99998999	0.84606841	0.84600395
180	0.99999499	0.84604746	0.84599540
181	1.00000000	0.84599975	0.84594476
182	0.99998999	0.84598336	0.84593174
183	0.99996063	0.84594807	0.84588915
184	1.00000000	0.84592941	0.84582505
185	0.99999008	0.84591907	0.84583986
186	0.99998501	0.84593514	0.84581825
187	0.99996999	0.84590071	0.84578407
188	0.99996501	0.84587952	0.84576884
189	0.99998501	0.84582567	0.84573334
190	0.99997067	0.84579670	0.84567308
191	0.99998999	0.84578952	0.84564203
192	0.99997506	0.84573105	0.84559438
193	0.99998999	0.84570247	0.84556815
194	0.99996999	0.84569108	0.84550563
195	0.99998069	0.84567350	0.84553504
196	0.99999070	0.84568036	0.84546959
197	0.99999499	0.84567553	0.84548214
198	1.00000000	0.84568834	0.84544203
199	0.99999499	0.84565824	0.84539443

This is the log file on your **FB15K** dataset:
Epoch	Valid	Test
0	0.21258732	0.21350736
1	0.34424557	0.34381460
2	0.47937773	0.47843648
3	0.56808072	0.56612557
4	0.62420285	0.62284073
5	0.66522732	0.66430059
6	0.69515529	0.69461825
7	0.71935973	0.71856740
8	0.73759723	0.73675600
9	0.75176904	0.75127017
10	0.76360717	0.76314065
11	0.77367082	0.77285811
12	0.78181231	0.78117460
13	0.78893596	0.78825158
14	0.79464367	0.79394436
15	0.79965779	0.79891652
16	0.80416784	0.80320734
17	0.80781418	0.80694491
18	0.81108990	0.81046605
19	0.81407723	0.81354842
20	0.81667709	0.81635964
21	0.81903249	0.81885853
22	0.82109931	0.82091731
23	0.82298642	0.82301471
24	0.82486853	0.82481173
25	0.82649991	0.82624820
26	0.82804829	0.82775813
27	0.82939535	0.82912040
28	0.83069190	0.83039460
29	0.83171308	0.83139932
30	0.83272809	0.83235934
31	0.83349121	0.83344638
32	0.83446184	0.83431569
33	0.83510509	0.83518907
34	0.83589664	0.83589852
35	0.83649659	0.83659431
36	0.83705834	0.83720976
37	0.83776906	0.83783305
38	0.83834016	0.83840394
39	0.83883491	0.83896181
40	0.83931002	0.83936596
41	0.83980522	0.83995181
42	0.84013960	0.84032023
43	0.84050202	0.84060919
44	0.84089246	0.84105113
45	0.84123555	0.84131080
46	0.84165716	0.84168682
47	0.84192261	0.84207213
48	0.84218460	0.84237254
49	0.84245339	0.84263423
50	0.84265190	0.84286454
51	0.84282124	0.84310189
52	0.84299335	0.84333399
53	0.84332195	0.84362227
54	0.84355536	0.84387347
55	0.84373599	0.84405529
56	0.84394130	0.84424263
57	0.84410715	0.84446669
58	0.84429693	0.84459850
59	0.84439135	0.84481570
60	0.84451100	0.84492943
61	0.84467703	0.84497029
62	0.84486023	0.84516272
63	0.84496894	0.84528482
64	0.84508294	0.84539589
65	0.84519145	0.84554064
66	0.84528682	0.84567285
67	0.84537861	0.84575692
68	0.84548092	0.84585649
69	0.84558696	0.84593722
70	0.84573057	0.84610561
71	0.84574851	0.84618628
72	0.84583104	0.84625101
73	0.84597796	0.84630573
74	0.84605747	0.84633306
75	0.84610146	0.84643668
76	0.84621009	0.84655190
77	0.84623879	0.84662759
78	0.84633097	0.84667701
79	0.84634042	0.84666130
80	0.84643418	0.84668970
81	0.84648758	0.84680939
82	0.84650791	0.84684694
83	0.84653121	0.84692636
84	0.84656152	0.84699786
85	0.84661219	0.84709558
86	0.84661749	0.84709668
87	0.84666082	0.84712788
88	0.84670296	0.84719133
89	0.84675634	0.84724995
90	0.84676403	0.84724417
91	0.84676820	0.84733453
92	0.84680352	0.84744552
93	0.84685972	0.84746245
94	0.84689054	0.84749284
95	0.84692457	0.84748578
96	0.84698904	0.84754845
97	0.84690079	0.84756041
98	0.84692711	0.84754947
99	0.84699202	0.84761971
100	0.84704185	0.84763774
101	0.84705469	0.84763399
102	0.84704277	0.84765586
103	0.84707248	0.84767416
104	0.84702805	0.84767899
105	0.84706312	0.84768814
106	0.84715536	0.84767562
107	0.84721795	0.84768403
108	0.84720945	0.84768522
109	0.84722167	0.84773076
110	0.84721404	0.84772056
111	0.84721702	0.84773859
112	0.84728742	0.84770960
113	0.84728926	0.84774402
114	0.84734976	0.84778410
115	0.84731501	0.84779674
116	0.84741566	0.84782439
117	0.84739107	0.84779879
118	0.84729117	0.84781134
119	0.84731171	0.84782970
120	0.84732980	0.84782100
121	0.84726146	0.84783986
122	0.84730437	0.84783909
123	0.84727165	0.84785762
124	0.84723085	0.84784967
125	0.84726924	0.84792364
126	0.84727404	0.84792477
127	0.84726635	0.84792799
128	0.84725103	0.84787238
129	0.84722188	0.84786892
130	0.84721172	0.84784093
131	0.84723634	0.84784409
132	0.84727401	0.84781635
133	0.84727201	0.84776321
134	0.84720200	0.84770805
135	0.84721854	0.84775108
136	0.84722400	0.84771210
137	0.84721756	0.84770700
138	0.84720501	0.84769720
139	0.84722731	0.84772941
140	0.84721914	0.84774718
141	0.84717855	0.84770057
142	0.84712976	0.84770173
143	0.84711733	0.84767681
144	0.84712321	0.84769174
145	0.84708938	0.84766129
146	0.84707320	0.84764475
147	0.84706855	0.84764397
148	0.84703904	0.84761724
149	0.84703434	0.84758499
150	0.84705076	0.84759974
151	0.84702227	0.84754214
152	0.84702975	0.84758264
153	0.84700632	0.84753498
154	0.84693855	0.84754840
155	0.84695029	0.84753019
156	0.84691596	0.84749040
157	0.84689823	0.84740123
158	0.84690204	0.84748235
159	0.84688139	0.84741437
160	0.84681758	0.84740335
161	0.84687313	0.84739602
162	0.84682325	0.84737438
163	0.84683150	0.84737250
164	0.84683901	0.84733865
165	0.84681138	0.84732494
166	0.84679842	0.84732980
167	0.84672916	0.84732679
168	0.84671387	0.84731510
169	0.84667236	0.84730572
170	0.84667927	0.84729102
171	0.84665594	0.84726185
172	0.84665683	0.84726265
173	0.84661898	0.84724274
174	0.84657836	0.84719962
175	0.84661990	0.84719303
176	0.84660980	0.84718305
177	0.84656027	0.84714520
178	0.84653687	0.84712970
179	0.84650657	0.84710813
180	0.84646201	0.84711751
181	0.84648508	0.84707522
182	0.84644800	0.84703219
183	0.84638324	0.84700710
184	0.84635654	0.84697950
185	0.84630013	0.84696463
186	0.84635559	0.84692952
187	0.84632835	0.84690070
188	0.84626627	0.84688720
189	0.84621182	0.84684193
190	0.84622246	0.84685332
191	0.84620795	0.84683934
192	0.84617031	0.84682143
193	0.84614643	0.84678918
194	0.84612688	0.84676951
195	0.84608319	0.84678707
196	0.84606808	0.84675702
197	0.84603623	0.84674743
198	0.84601948	0.84670722
199	0.84600386	0.84666497

**Comment by timlacroix** [comment]
oops. great catch ! I think this comes from a difference in default values for the init scale of embeddings between this code and my grid search set-up.

I'll make sure that setting it to 1e-3 fixes everything before closing this issue ! 

As far as I know, all the version of the FB15K dataset you mentioned are exactly the same, up to relabelling of the entities.

**Comment by timlacroix** [comment]
After re-checking, I get 0.8556 MRR onFB15K with an init size of 1e-3. I'll update the defaults in the code and add init-size to the README. Thanks again for catching this.

**Comment by phucty** [comment]
Great
Thank you very much for your time and effort.
I have also obtained the same result as you mention. 
Best,
